modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/EncryptedFileIO.java

```diff
@@ -109,7 +109,7 @@
 
     /** {@inheritDoc} */
     @Override public int read(ByteBuffer destBuf) throws IOException {
-        assert position() == 0;
+        assert position() == 0 && headerSize > 0;
 
         return plainFileIO.read(destBuf);
```

Comment:
We don't decrypt because implementation on this FileIO deprecated chipered reads for data with offsets 0. You can notice `assert position() == 0` before this PR. Same here, we deprecate chipered reads. Why? This question is about design of EncryptedFileIO. Surely, it requires some refactoring, revising all its methods. See
```
@Override public int read(byte[] buf, int off, int len) throws IOException {
        throw new UnsupportedOperationException("Encrypted File doesn't support this operation");
    }
```
Same here. However, refactoring of EncriptedFileIO is definitely out of scope of this ticket.